crypto/ex_data.c

```diff
@@ -287,8 +287,6 @@ int CRYPTO_dup_ex_data(int class_index, CRYPTO_EX_DATA *to,
         CRYPTOerr(CRYPTO_F_CRYPTO_DUP_EX_DATA, ERR_R_MALLOC_FAILURE);
         return 0;
     }
-    if (!CRYPTO_set_ex_data(to, mx - 1, NULL))
-        goto err;
```

Comment:
I have never seen any use case of the new callback, I must admit,
especially because the new function gets the value as an in parameter.

... and it has no way to return an error.

IMHO the function signature makes this function
un-usable.